dl/dl.go

```diff
@@ -0,0 +1,30 @@
+// Copyright 2022 The Ebiten Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//go:build darwin
+// +build darwin
+
+package dl
+
+import (
+	"github.com/ebiten/purego/syscall"
+	"unsafe"
+)
+
+const RTLD_DEFAULT = ^uintptr(1)
+
+func Sym(handle uintptr, name *byte) uintptr {
+	ret, _, _ := syscall.SyscallX(dlsymABI0, handle, uintptr(unsafe.Pointer(name)), 0)
```

Comment:
Are we happy with this signature? Would we rather it take a string and convert it into a *byte inside of Sym?

Reply:
`string` is better for the `name`. We can write the code like this:

```go
bs := []byte(name)
ret, _, _ := syscall.SyscallX(dlsymABI0, handle, uintptr(unsafe.Pointer(&bs[0])), 0)
runtime.KeepAlive(bs)
```

Reply:
I realized we have to care about the NUL termination. Then

```
bs := []byte(name + "\x00")
ret, _, _ := syscall.SyscallX(dlsymABI0, handle, uintptr(unsafe.Pointer(&bs[0])), 0) 
runtime.KeepAlive(bs)
```

Reply:
I went ahead and added `_CString` which returns a C string with null termination if it doesn't already have one.